pkg/controller/sparkapplication/job.go

```diff
@@ -30,62 +35,96 @@ import (
 	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/config"
 )
 
+const (
+	sparkSubmitPodMemory = "1000Mi"
+	sparkSubmitPodCpu    = "1024m"
+)
+
+type submissionJobManagerIface interface {
+	createSubmissionJob(app *v1beta1.SparkApplication) (*string, *string, error)
+	deleteSubmissionJob(app *v1beta1.SparkApplication) error
+	getSubmissionJob(app *v1beta1.SparkApplication) (*batchv1.Job, error)
+	hasJobSucceeded(app *v1beta1.SparkApplication) (*bool, *metav1.Time, error)
```

Comment:
Use `bool` instead of `*bool`.

Reply:
This already existed and we need a pointer as we need to return a nil value if we weren't able to query the status at all ?